tests/unit/cartography/graph/test_querybuilder_complex.py

```diff
@@ -0,0 +1,139 @@
+    lastupdated: PropertyRef = PropertyRef('lastupdated', set_in_kwargs=True)
+    another_rel_field: PropertyRef = PropertyRef('AnotherField')
+    yet_another_rel_field: PropertyRef = PropertyRef("YetAnotherRelField")
+
+
+@dataclass
+class InterestingAssetToSubResourceRel(CartographyRelSchema):
+    """
+    Define a sub resource relationship
+    (:InterestingAsset)<-[:RELATIONSHIP_LABEL]-(:SubResource)
+    """
+    target_node_label: str = 'SubResource'
+    target_node_key: str = 'id'
+    target_node_key_property_ref: PropertyRef = PropertyRef('sub_resource_id', set_in_kwargs=True)
+    direction: LinkDirection = LinkDirection.INWARD
+    rel_label: str = "RELATIONSHIP_LABEL"
+    properties: InterestingAssetToSubResourceRelProps = InterestingAssetToSubResourceRelProps()
+
+
+@dataclass
+class InterestingAssetToHelloAssetRelProps(CartographyRelProperties):
+    lastupdated: PropertyRef = PropertyRef('lastupdated', set_in_kwargs=True)
+
+
+@dataclass
+class InterestingAssetToHelloAssetRel(CartographyRelSchema):
+    """
+    Define an additional relationship
+    (:InterestingAsset)-[:ASSOCIATED_WITH]->(:HelloAsset)
+    """
+    target_node_label: str = 'HelloAsset'
+    target_node_key: str = 'id'
+    target_node_key_property_ref: PropertyRef = PropertyRef('hello_asset_id')
+    direction: LinkDirection = LinkDirection.OUTWARD
+    rel_label: str = "ASSOCIATED_WITH"
+    properties: InterestingAssetToHelloAssetRelProps = InterestingAssetToHelloAssetRelProps()
+
+
+@dataclass
+class InterestingAssetToWorldAssetRelProps(CartographyRelProperties):
+    lastupdated: PropertyRef = PropertyRef('lastupdated', set_in_kwargs=True)
+
+
+@dataclass
+class InterestingAssetToWorldAssetRel(CartographyRelSchema):
+    """
+    Define yet another relationship.
+    (:InterestingAsset)<-[:CONNECTED]-(:WorldAsset)
+    """
+    target_node_label: str = 'WorldAsset'
+    target_node_key: str = 'id'
+    target_node_key_property_ref: PropertyRef = PropertyRef('world_asset_id')
+    direction: LinkDirection = LinkDirection.INWARD
+    rel_label: str = "CONNECTED"
+    properties: InterestingAssetToWorldAssetRelProps = InterestingAssetToWorldAssetRelProps()
+
+
+@dataclass
+class InterestingAssetSchema(CartographyNodeSchema):
+    extra_labels: Optional[List[str]] = default_field(['AnotherNodeLabel', 'YetAnotherNodeLabel'])
+    label: str = 'InterestingNode'
+    properties: SimpleNodeProperties = SimpleNodeProperties()
+    sub_resource_relationship: InterestingAssetToSubResourceRel = InterestingAssetToSubResourceRel()
+    other_relationships: Optional[List[CartographyRelSchema]] = default_field(
+        [
+            InterestingAssetToHelloAssetRel(),
+            InterestingAssetToWorldAssetRel(),
+        ],
+    )
+
+
+def test_build_ingestion_query_complex():
+    # Act
+    query = build_ingestion_query(InterestingAssetSchema())
+
+    expected = """
+        UNWIND $DictList AS item
```

Comment:
@csanders-git brought up a good point - how do we handle the case where a given node only attaches to another given node _sometimes_ -- depending on the order, a null match will cause the other links to not be created.

Reply:
Then we might use the `OPTIONAL MATCH`, but I'm not 100% sure yet.

Reply:
I figured out how to do this using `CALL{ query1 UNION query2 UNION ... queryN }`